realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -533,6 +536,82 @@ boolean removeAllInternal(Collection<?> c) {
     }
 }
 
+/**
+ * TODO
+ */
+class UUIDOperator extends SetValueOperator<UUID> {
+
+    public UUIDOperator(BaseRealm baseRealm, OsSet osSet, Class<UUID> valueClass) {
+        super(baseRealm, osSet, valueClass);
+    }
+
+    @Override
+    boolean add(@Nullable UUID value) {
+        return osSet.add(value);
+    }
+
+    @Override
+    boolean contains(@Nullable Object o) {
+        UUID value;
+        if (o == null) {
+            value = null;
+        } else {
+            value = (UUID) o;
+        }
+        return osSet.contains(value);
+    }
+
+    @Override
+    boolean remove(@Nullable Object o) {
+        return osSet.remove((UUID) o);
+    }
+
+    @Override
+    boolean containsAll(Collection<?> c) {
+        if (isRealmCollection(c)) {
+            OsSet otherOsSet = ((RealmSet<?>) c).getOsSet();
+            return funnelCollection(otherOsSet, OsSet.ExternalCollectionOperation.CONTAINS_ALL);
+        }
+
+        for (Object value : c) {
+            if (value != null && !(value instanceof UUID)) {
+                return false;
+            }
+        }
+
+        //noinspection unchecked
+        Collection<UUID> uuidCollection = (Collection<UUID>) c;
+        NativeMixedCollection collection = NativeMixedCollection.newUUIDCollection(uuidCollection);
+        return osSet.collectionFunnel(collection, OsSet.ExternalCollectionOperation.CONTAINS_ALL);
+    }
+
+    @Override
+    boolean addAllInternal(Collection<? extends UUID> c) {
+        for (Object value : c) {
+            if (value != null && !(value instanceof UUID)) {
+                throw new IllegalArgumentException("Invalid collection type. Set and collection must contain the same type of elements.");
+            }
+        }
+
+        NativeMixedCollection collection = NativeMixedCollection.newUUIDCollection(c);
+        return osSet.collectionFunnel(collection, OsSet.ExternalCollectionOperation.ADD_ALL);
+    }
+
+    @Override
+    boolean removeAllInternal(Collection<?> c) {
+        //noinspection unchecked
```

Comment:
We are also not guaranteed to have an UUID collection here, aren't we?

Similarly for all the other collection methods.

Reply:
Fixed.